File: opcua-webservice-spring/examples/uawebservicetest/src/main/java/org/opcfoundation/uawebservicetest/testdigitaltwin/EventTestType.java
package org.opcfoundation.uawebservicetest.testdigitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.addressspace.nodes.builtin.UaDataTypes;
import org.opcfoundation.webserver.digitaltwin.DigitalTwinSpace;
import org.opcfoundation.webserver.digitaltwin.event.EventData;
import org.opcfoundation.webserver.digitaltwin.event.EventType;

import java.time.LocalDateTime;

public class EventTestType extends EventType {
    public static final String Customized = "Customized";

    public EventTestType(DigitalTwinSpace space)
    {
        super("EventTestType", new LocalizedText("EventTestType"), space);
        addField(Customized, new LocalizedText(Customized), new LocalizedText("Customized field"), UaDataTypes.String);
    }

    public static EventData generateEventData(
            String eventId,
            LocalDateTime time,
            String message,
            String customized)
    {
        EventData data = new EventData("eventId");
        data.setTime(time);
        data.setMessage(message);
        data.setFieldData(Customized, Variant.ofString(customized));

        return data;
    }
}

File: opcua-webservice-node/examples/testserver/testdigitaltwin/EventTestType.ts
import { UaLocalizedText, UaVariant } from "opcua-webapi-ts";
import { UaDataTypes } from "../../../src/server/addressspace/nodes";
import { DigitalTwinSpace } from "../../../src/server/digitaltwin";
import { EventData, EventType } from "../../../src/server/digitaltwin/event";

export class EventTestType extends EventType {
    static readonly Customized = "Customized";

    constructor(space: DigitalTwinSpace) {
        super("EventTestType", new UaLocalizedText("EventTestType"), space);
        this.description = new UaLocalizedText("EventTestType");

        this.addField(
            EventTestType.Customized,
            new UaLocalizedText(EventTestType.Customized),
            new UaLocalizedText("Customized field"),
            UaDataTypes.String,
        );
    }

    static generateEventData(
        eventId: string,
        time: Date,
        message: string,
        customized: string,
    ): EventData {
        const data = new EventData(eventId);
        data.time = time;
        data.message = message;
        data.setFieldData(EventTestType.Customized, UaVariant.string(customized));
        return data;
    }
}